dev/bots/analyze.dart

```diff
@@ -435,7 +435,7 @@ Future<void> verifyGoldenTags(String workingDirectory, { int minimumMatches = 20
 final RegExp _findDeprecationPattern = RegExp(r'@[Dd]eprecated');
 final RegExp _deprecationPattern1 = RegExp(r'^( *)@Deprecated\($'); // flutter_ignore: deprecation_syntax (see analyze.dart)
 final RegExp _deprecationPattern2 = RegExp(r"^ *'(.+) '$");
-final RegExp _deprecationPattern3 = RegExp(r"^ *'This feature was deprecated after v([0-9]+)\.([0-9]+)\.([0-9]+)(\-[0-9]+\.[0-9]+\.pre)?\.',?$");
+final RegExp _deprecationPattern3 = RegExp(r"^ *'This feature was deprecated after v(\d+)\.(\d+)\.(\d+)(-\d+\.\d+\.pre)?\.',?$");
```

Comment:
This is more readable.  Personally, I'd add names to the capture groups, like so:

```suggestion
final RegExp _deprecationPattern3 = RegExp(r"^ *'This feature was deprecated after v(?<major>\d+)\.(?<minor>\d+)\.(?<patch>\d+)(?<build>-\d+\.\d+\.pre)?\.',?$");
```

And then refer to them with those names (with `match.namedGroup('major')`, etc.) instead of indices in the code.

Reply:
And, I can't leave comments on the right lines (thanks GitHub!), but I'd name the match objects something better than `match1`, `match2`, etc.  Their types should be changed to [`RegExpMatch`](https://api.flutter.dev/flutter/dart-core/RegExpMatch-class.html) too, which is the newer interface that has `namedGroup` on it.

Reply:
Same with `_deprecationPattern1`, `_deprecationPattern2`, etc.

Reply:
Done.  However, it looks like there is a bug in IntelliJ/Android Studio that reports that the 'group syntax is not supported in this dialect' (see [issue](https://youtrack.jetbrains.com/issue/IDEA-215461)). It works, but shows as errors in the IDE.

That said, the version groups are no longer checked directly (just that they match the pattern), so I am not sure how important this is.

I also renamed a bunch of these patterns and the matches to make them more readable.

Reply:
That's a shame. I'm sure it doesn't show as an error in Emacs. :-)